core-implementation/src/main/java/gg/solarmc/loader/impl/Icarus.java

```diff
@@ -19,25 +19,26 @@
 
 package gg.solarmc.loader.impl;
 
+import gg.solarmc.loader.impl.launch.DataGroup;
+import gg.solarmc.loader.impl.login.LoginHandlerBuilderImpl;
+
+import java.util.Set;
+
 public class Icarus implements AutoCloseable {
 
-	private final LoginHandler loginHandler;
 	private final TransactionSource transactionSource;
 	private final DataManagementCenter dataManagement;
+	private final Set<DataGroup<?, ?, ?>> groups;
 	private final DataCenterLifecycle lifecycle;
 
-	Icarus(LoginHandler loginHandler, TransactionSource transactionSource,
-		   DataManagementCenter dataManagement, DataCenterLifecycle lifecycle) {
-		this.loginHandler = loginHandler;
+	Icarus(TransactionSource transactionSource, DataManagementCenter dataManagement,
```

Comment:
Why are you moving datagroups into the icarus?

Reply:
The LoginHandler.Builder needs them. See line 56